File: apps/server/server/Models/Appointment.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ZhmApi.Models
{
    public enum AppointmentStatus
    {
        PendingAccess,          // appointment exists, waiting on access
        Scheduled,              // access approved, operation allowed
        AccessDenied,           // access denied, appointment blocked
        Completed,
        Cancelled
    }
    public class Appointment
    {
        public int Id { get; set; }

        public int ReferralId { get; set; }
        [ForeignKey("ReferralId")]
        public Referral Referral { get; set; } = null!;
        public int SpecialistId {get; set;}
        public User Specialist {get; set;} = null!;
        public DateTime Date {get; set;}
        public AppointmentStatus Status { get; set; } = AppointmentStatus.Scheduled;
    }
}

File: apps/client/src/types/appointment.ts
export type AppointmentStatus =
  | 'PendingAccess'
  | 'Scheduled'
  | 'AccessDenied'
  | 'Completed'
  | 'Cancelled'

export interface AppointmentDto {
  id: number
  referralId: number
  notes: string
  status: AppointmentStatus
  treatmentDescription: string
  treatmentInstructions?: string
  patientName: string
  date: string
}
export interface AppointmentCreateDto {
  referralId: number
  specialistId: number
  date: string
  time: string
}